frontend/src/bbkit/BBProgressPie.vue

```diff
@@ -0,0 +1,74 @@
+<template>
+  <div class="relative flex justify-center items-center">
+    <svg :viewBox="viewBox" class="absolute inset-0 z-0">
+      <path
+        class="stroke-control-bg"
+        fill="none"
+        :stroke-width="thickness + 1"
+        d="M18 2.0854 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
+      />
+      <path
+        fill="none"
+        stroke="currentColor"
+        stroke-linecap="round"
+        animation="progress 1s ease-out forwards"
+        :stroke-width="thickness"
+        :stroke-dasharray="strokeDasharray"
+        d="M18 2.0854 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
+      />
+    </svg>
+    <slot name="default" :percent="displayPercent">
+      {{ displayPercent }}%
+    </slot>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { computed, withDefaults } from "vue";
+
+/**
+ * [Learn more](https://medium.com/@pppped/how-to-code-a-responsive-circular-percentage-chart-with-svg-and-css-3632f8cd7705)
+ */
+
+const props = withDefaults(
+  defineProps<{
+    percent?: number; // range: [0, 100]
+    thickness?: number; // min value: 1
+  }>(),
+  {
+    percent: 0,
+    thickness: 3,
+  }
+);
+
+const viewBox = computed(() => {
+  const offset = Math.max(props.thickness - 3, 0);
```

Comment:
extract the `3` as a const variable and comment it?

Reply:
we are somehow limited by vue compiler